### ISSUE TITLE
listener: repl blocks when long process runs

### ISSUE THREAD

**Issue by dkochmanski** [robert-strandh/McCLIM#307]
When we start long operation, whole repl is blocked and not repainted when screen is changed

Steps to reproduce:
1. start the listener
2. run `(sleep 120)`
3. switch virtual desktop (like in xfce) back and forth
4. notice window is not repainted (it's empty) until command finishes

**Comment by dkochmanski** [comment]
in dev-commands.lisp:

```
;;; The background evaluation feature is neat, but there are thread
;;; safety issues doing output to streams, special variables have
;;; unexpected values, and input doesn't work right due to racing to
;;; read from the event queue. Sadly, I am forced to disable it by
;;; default.

(defparameter *use-background-eval* nil
  "Perform evaluation in a background thread, which can be interrupted.")
```

we should find a way around this (custom read? event-queue tinkering?) -  calling `read` when we have this param set to T indeed breaks badly.

**Comment by dkochmanski** [comment]
This issue should be solved by providing unique streams for the Listener evaluated requests which get piped back and forth to the Listener frame, so there are no races and simple eval doesn't share the same stream with the Listener.

**Comment by krant-avent** [comment]
Hm. read, print and clim:accept work correctly in both foreground an background modes. May be comment outdated?

**Comment by dkochmanski** [comment]
no, I can see problems, when I try to `read` with background mode. There is a race for key presses between CLIM event loop and function `cl:read`.

**Comment by dkochmanski** [comment]
Probably way to tackle it would be binding a separate stream for expression input and making McCLIM forward it there. Same goes for other default streams.

**Comment by goose121** [comment]
Would it potentially be an issue that binding a separate stream for the output would mean that the output records of the listener pane wouldn't be accessible to the evaluated code? I can see that potentially messing up `accepting-values` and possibly `updating-output`
